geopandas/plotting.py

```diff
@@ -552,35 +563,50 @@ def plot_dataframe(
     point_idx = np.asarray((geom_types == "Point") | (geom_types == "MultiPoint"))
 
     # plot all Polygons and all MultiPolygon components in the same collection
-    polys = df.geometry[poly_idx]
+    polys = df.geometry[poly_idx * np.invert(nan_idx)]
     if not polys.empty:
         plot_polygon_collection(
-            ax, polys, values[poly_idx], vmin=mn, vmax=mx, cmap=cmap, **style_kwds
+            ax,
+            polys,
+            values[poly_idx * np.invert(nan_idx)],
```

Comment:
Suggestion: maybe do like `subset = values[poly_idx * np.invert(nan_idx)]` on the line before, to avoid that black puts those argument all on separate lines)